scripts/xtensa-build-zephyr.py

```diff
@@ -279,8 +280,26 @@ def west_init_update():
 	link back to sof/"""
 	global west_top, SOF_TOP
 	zephyr_dir = pathlib.Path(west_top, "zephyr")
-	execute_command(["git", "clone", args.url, str(zephyr_dir)], check=True, timeout=1200)
-	execute_command(["git", "fetch", "origin", args.z], check=True, timeout=300, cwd=zephyr_dir)
+
+	z_remote = "origin"
+	z_ref = args.zephyr_ref
```

Comment:
This variable is not needed. We can write directly to args.zephyr_ref. Value will not change as it is already after parse_args() call. But even this is not needed as you can just add this comments as a help message to argparse new argument what I suggested above.

Reply:
args.zephyr_ref is a global variable, this is a local one. Immutability is "best practice", especially for global variables. In some newer languages variables are immutable by default.

https://www.bing.com/search?q=immutability%20concurrency
https://queue.acm.org/detail.cfm?id=3212479
>  The cache coherency protocol is one of the hardest parts of a modern CPU to make both fast and correct. Most of the complexity involved comes from supporting a language in which data is expected to be both shared and mutable as a matter of course

Etc.

Of course there's approximately zero chance this script will ever be multithreaded but less global and less mutability is just good practice and who knows: maybe some other function could want to use `args.zephyr_ref` in the future.

Reply:
Yeah, this made sense only with my previous remark if it would be actually implemented. Dropping.